### DISCUSSION TITLE
Extending path/railings/supports

### DISCUSSION THREAD

**Discussion by dr-dirt** [OpenRCT2/OpenRCT2#16035] [Feature requests]
**Colorable Paths**
Extending the options for paths to allow them to be recolored.  This would save a number of path spots and Brown/Green/Red tarmac paths can just be reduced to a single “Tarmac” for example.

**Recolorable Rails/Supports**
Same idea, reduces the need for multiple objects that are the same design but different colors, also allows you to change the bright white queue fences which might be undesirable.

**Footpath Textures as Queues**
I propose an option to use the regular footpath texture as a queue texture.  I had created several objects that are just that, but seems quite superfluous when the images are all there to have them used as queues.  This would probably mean the IsQueue flag to represent that it is _only_ a queue.

**Load Legacy paths as Surface and Railings**
Don’t know then plausibility of this, but currently it’s a bit awkward to have so many tabs to select paths from; I would think it would be ideal to load legacy DATs to the new formats for path/rails.  The issue I could see coming up is multiple duplicate path supports or even textures as the old way to do paths required individual DATs for any change to those.

**Consistent/updated previews**
I noticed the queue preview images are slightly different from each other and also don’t reflect the selected supports/railings. It’s easy enough to create the preview images manually but I think something automatically generated is ideal.  It would be very nice to have the images rendered with the railing and supports that you’ve selected, so it ensures the preview images are all uniform.  This could also make it possible to show the supports on the footpath surfaces as well like how they used to show this in the corner.

**Comment by danielmeinert** [comment]
While we are at it, also colorable path surfaces would be cool to complete the variability.

**Reply by dr-dirt** [reply]
Updated this as I think this is a good point and there’s a few other areas that would be nice to expand on.

**Comment by J-PIE-314** [comment]
Most of the requests you suggested here have been implemented now in one way or another except for "Colorable Paths" and "Recolorable Rails/Supports" which may go well as a part of this suggestion: #20383.

**Comment by J-PIE-314** [comment]
Color-able paths have been suggested in #15889.
